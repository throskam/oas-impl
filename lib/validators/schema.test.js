/* globals describe, it, expect */

const createSchemaValidator = require('./schema')

describe('Type', () => {
  const schema = {
    type: 'integer'
  }

  const validator = createSchemaValidator(schema)

  it('should return no errors when the value is valid', () => {
    const payload = { value: 1 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return validation error when the value is invalid', () => {
    const payload = { value: 'string' }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return validation error when the value is missing', () => {
    expect(validator()).toMatchSnapshot()
  })
})
