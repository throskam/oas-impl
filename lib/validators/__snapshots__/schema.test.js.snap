// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type should return no errors when the value is valid 1`] = `Array []`;

exports[`Type should return validation error when the value is invalid 1`] = `
Array [
  Object {
    "ajv": Object {
      "dataPath": "",
      "keyword": "type",
      "message": "should be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "schema",
    "rule": "schema-invalid",
    "value": "string",
  },
]
`;

exports[`Type should return validation error when the value is missing 1`] = `
Array [
  Object {
    "ajv": Object {
      "dataPath": "",
      "keyword": "type",
      "message": "should be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "schema",
    "rule": "schema-invalid",
    "value": undefined,
  },
]
`;
